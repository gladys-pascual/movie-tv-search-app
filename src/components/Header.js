import React from "react";
import { Link, NavLink } from "react-router-dom";

const Header = ({ onLogIn, userDetails }) => {
  return (
    <header>
      <Link exact to="/" className="logo">
        <img src={process.env.PUBLIC_URL + "logo.png"} alt="logo" />
      </Link>
      <nav>
        <ul>
          <li>
            <NavLink exact to="/" activeClassName="selected">
              Home
            </NavLink>
          </li>
          <li>
            {userDetails.username && (
              <NavLink to="/favorites" activeClassName="selected">
                My Favourites
              </NavLink>
            )}
          </li>
          <li>
            {userDetails.username ? (
              <button className="username">{userDetails.username}</button>
            ) : (
              <button onClick={onLogIn} className="log-in">
                Login
              </button>
            )}
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
