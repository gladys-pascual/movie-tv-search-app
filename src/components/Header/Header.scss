@import "../../assets/variables.scss";

header {
  margin: 30px 50px 50px 50px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  @media screen and (max-width: 780px) {
    margin: 10px 10px 30px 10px;
  }
  .logo {
    width: 150px;
    img {
      width: 100%;
    }
  }
  nav ul {
    display: flex;
    padding: 20px;
    margin: 4px;
    font-size: 16px;
    list-style-type: none;
    @media screen and (max-width: 780px) {
      padding: 10px;
    }

    a {
      text-decoration: none;
      color: $text-color;
    }
    .selected {
      color: $text-color-hover;
      background-color: $theme-color;
      padding: 12px;
      border-radius: 8px;
    }
    .dropdown {
      position: relative;
      display: inline-block;

      .dropdown-content {
        display: none;
        position: absolute;
        min-width: 200px;
        li {
          margin: 12px 20px;
        }
      }

      .dropdown-content a:hover {
        color: $text-color-hover;
        background-color: $theme-color;
        padding: 4px;
        border-radius: 4px;
      }

      .log-in,
      .dropbtn {
        border: none;
        background: none;
        color: $text-color;
        cursor: pointer;
        padding: 0px 20px;
      }
      .username {
        font-weight: 600;
        display: flex;
        align-items: center;
      }
    }
    .dropdown:hover .dropdown-content {
      display: block;
    }
  }
}
