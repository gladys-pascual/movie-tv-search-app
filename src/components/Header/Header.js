import React from "react";
import { Link, NavLink } from "react-router-dom";
import "./Header.scss";

const Header = ({ onLogIn, userDetails, onLogOut }) => {
  return (
    <header>
      <Link exact="true" to="/" className="logo">
        <img src={process.env.PUBLIC_URL + "logo.png"} alt="logo" />
      </Link>
      <nav>
        <ul>
          <li>
            <NavLink exact={true} to="/" activeClassName="selected">
              Home
            </NavLink>
          </li>

          <div className="dropdown">
            {userDetails.username ? (
              <button className="username dropbtn">
                <span>{userDetails.username}</span>
                <span className="material-icons">arrow_drop_down</span>
              </button>
            ) : (
              <button onClick={onLogIn} className="log-in dropbtn">
                Login
              </button>
            )}

            <div className="dropdown-content">
              <li>
                {userDetails.username && (
                  <Link to="/favorites">My Favourites</Link>
                )}
              </li>

              <li>
                {userDetails.username && (
                  <Link exact="true" to="/" onClick={onLogOut}>
                    Logout
                  </Link>
                )}
              </li>
            </div>
          </div>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
