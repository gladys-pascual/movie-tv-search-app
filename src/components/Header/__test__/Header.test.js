import React from "react";
import { render } from "@testing-library/react";
import Header from "../Header";
import { BrowserRouter } from "react-router-dom";

test("render Header component when no user is logged in as expected", () => {
  const props = {
    onLogIn: jest.fn(),
    userDetails: {},
  };
  const { container } = render(
    <BrowserRouter>
      <Header {...props} />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});

test("render Header component when there is a user logged in as expected", () => {
  const props = {
    userDetails: { username: "gladyskate" },
    onLogOut: jest.fn(),
  };
  const { container } = render(
    <BrowserRouter>
      <Header {...props} />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
