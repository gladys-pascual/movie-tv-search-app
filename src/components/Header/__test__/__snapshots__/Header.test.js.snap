// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header component when no user is logged in as expected 1`] = `
<div>
  <header>
    <a
      class="logo"
      exact="true"
      href="/"
    >
      <img
        alt="logo"
        src="logo.png"
      />
    </a>
    <nav>
      <ul>
        <li>
          <a
            aria-current="page"
            class="selected"
            href="/"
          >
            Home
          </a>
        </li>
        <div
          class="dropdown"
        >
          <button
            class="log-in dropbtn"
          >
            Login
          </button>
          <div
            class="dropdown-content"
          >
            <li />
            <li />
          </div>
        </div>
      </ul>
    </nav>
  </header>
</div>
`;

exports[`render Header component when there is a user logged in as expected 1`] = `
<div>
  <header>
    <a
      class="logo"
      exact="true"
      href="/"
    >
      <img
        alt="logo"
        src="logo.png"
      />
    </a>
    <nav>
      <ul>
        <li>
          <a
            aria-current="page"
            class="selected"
            href="/"
          >
            Home
          </a>
        </li>
        <div
          class="dropdown"
        >
          <button
            class="username dropbtn"
          >
            <span>
              gladyskate
            </span>
            <span
              class="material-icons"
            >
              arrow_drop_down
            </span>
          </button>
          <div
            class="dropdown-content"
          >
            <li>
              <a
                href="/favorites"
              >
                My Favourites
              </a>
            </li>
            <li>
              <a
                exact="true"
                href="/"
              >
                Logout
              </a>
            </li>
          </div>
        </div>
      </ul>
    </nav>
  </header>
</div>
`;
